pkg/file/file.go

```diff
@@ -25,14 +25,26 @@ package file
 
 import (
 	"path/filepath"
+	"strings"
 )
 
-// ExtensionIn returns true if the file's extension is one of the provided ones
-func ExtensionIn(path string, extensions []string) bool {
+// HasExtension returns true if the file's extension is one of the provided ones
+func HasExtension(path string, extensions []string) bool {
 	for _, ext := range extensions {
 		if filepath.Ext(path) == ext {
 			return true
 		}
 	}
 	return false
 }
+
+// ShouldIgnore returns true if the path matches any of the ignore strings
+func ShouldIgnore(path string, ignoreFolders []string) bool {
+	dir, _ := filepath.Split(path)
+	for _, igfolder := range ignoreFolders {
+		if strings.Contains(dir, igfolder) {
```

Comment:
if "igfolder" is a part of the name of any folder of the path "dir", this method will fail,
consider different impl, ej:

```
func shouldIgnore(path string, ignoreFolders []string) bool {
	segments := strings.Split(path, string(os.PathSeparator))
	ignoreLen := len(ignoreFolders)
	for _, segment := range segments {
		fmt.Println(segment)
		if sort.Search(ignoreLen, func(i int) bool { return ignoreFolders[i] == segment }) < ignoreLen {
			return true
		}
	}
	return false
}

func main() {
	ignores := []string{"ok", "demo1", "ok1"}
	sort.Strings(ignores)

	res := shouldIgnore("c:\\demo\\ok\\file.txt", ignores)
	fmt.Println(res)
}
```

Reply:
Well spotted. Added tests to control this specific use case. I used a nested for as I find it more readable (and not necessarily less performant) than the sort.Search() solution.